### ISSUE TITLE
推理出现无端端的英文，怎么处理呢

### ISSUE THREAD

**Issue by deeplearning-LI** [QwenLM/Qwen2#766]
我本来是中文问答的任务，回答到最后，出现类似下面这种规整的乱码。。。：
If $x$ is a real number and $x^2 = 16$, what is the sum of all possible values of $x$? The equation $x^2 = 16$ has two solutions: $x = 4$ and $x = -4$.
The sum of these two solutions is $4 + (-4) = \boxed{0}$.

**Comment by jklj077** [comment]
Hi, the English response looks good. 

I need more info, models, frameworks, cards, steps to reproduce...

**Comment by deeplearning-LI** [comment]
1-推理模型加载代码：
class Qwen2_LLM(LLM):
    # 基于本地 Qwen2 自定义 LLM 类
    tokenizer: AutoTokenizer = None
    model: AutoModelForCausalLM = None
    def __init__(self, mode_name_or_path : str):

        super().__init__()
        print("loading local model ....")
        self.tokenizer = AutoTokenizer.from_pretrained(mode_name_or_path, 
                                                       pad_token='<|extra_0|>', eos_token='<|endoftext|>', padding_side='left', trust_remote_code=True, use_fast=False
                                                       )
        self.model = AutoModelForCausalLM.from_pretrained(mode_name_or_path, 
                                                          torch_dtype=torch.bfloat16, device_map="auto", 
                                                          )
        self.model.generation_config = GenerationConfig.from_pretrained(mode_name_or_path)
        self.model = self.model.eval()
        print("完成本地模型的加载")
    
    def _call(self, prompt : str, stop: Optional[List[str]] = None,
                run_manager: Optional[CallbackManagerForLLMRun] = None,
                **kwargs: Any):
        # qwen2 genetated
        messages = [{"role": "user","content": prompt }]
        input_ids = self.tokenizer.apply_chat_template(messages, tokenize=False, add_generation_prompt=True)
        model_inputs = self.tokenizer([input_ids], return_tensors="pt").to('cuda')
        generated_ids = self.model.generate(model_inputs.input_ids, 
                                            temperature = 0.2, 
                                            max_new_tokens=512, 
                                            top_p=0.9, 
                                            top_k=40, 
                                            eos_token_id=self.tokenizer.eos_token_id)
        input_text = self.tokenizer.batch_decode(model_inputs.input_ids, skip_special_tokens=True)[0]
        response = self.tokenizer.batch_decode(generated_ids, skip_special_tokens=True)[0]
        response = response.split(input_text)[-1].strip()
        remove_str = ['<|im_end|>','<|im_end|>','<|im_start|>','<|im_start|>','user','assistant']
        for i in remove_str:
            response = response.replace(i,'')
        
        return response
    @property
    def _llm_type(self) -> str:
        return "Qwen2_LLM"

我想结合langchain做RAG推理的。

2-提问问题后，推理得到的结果是这样的：
![image](https://github.com/QwenLM/Qwen2/assets/86917492/d9f8cd1a-2491-42c1-a5c7-18ff2e8c7608)
![image](https://github.com/QwenLM/Qwen2/assets/86917492/7dc90943-1dc3-41f5-9234-bdc9513a63cb)

不知道咋解决了，终端报了提示是这样的：

“UserWarning: `do_sample` is set to `False`. However, `top_k` is set to `40` -- this flag is only used in sample-based generation modes. You should set `do_sample=True` or unset `top_k`.
  warnings.warn(
The attention mask and the pad token id were not set. As a consequence, you may observe unexpected behavior. Please pass your input's `attention_mask` to obtain reliable results.
Setting `pad_token_id` to `eos_token_id`:151643 for open-end generation.
The attention mask is not set and cannot be inferred from input because pad token is same as eos token.As a consequence, you may observe unexpected behavior. Please pass your input's `attention_mask` to obtain reliable results.”

3-顺便请教一下，Qwen2-7b怎么加载快一些，我单卡4090加载（大约3分钟）后还要等一段时间才能完全加载（又大约3分），这可怎么办。感谢~

十分感谢回复~

**Comment by jklj077** [comment]
Hi, it appears that you were you using Qwen2-7B. If so, please use Qwen2-7B-Instruct.

**Comment by deeplearning-LI** [comment]
好吧~正在下载~感谢
